lib/charms/mysql/v0/mysql.py

```diff
@@ -1513,9 +1514,29 @@ def get_innodb_buffer_pool_parameters(self) -> Tuple[int, Optional[int]]:
                 innodb_buffer_pool_chunk_size = chunk_size
 
             return (pool_size, innodb_buffer_pool_chunk_size)
-        except Exception as e:
-            logger.exception("Failed to compute innodb buffer pool parameters", exc_info=e)
-            raise MySQLGetInnoDBBufferPoolParametersError("Error retrieving total free memory")
+        except Exception:
+            logger.exception("Failed to compute innodb buffer pool parameters")
+            raise MySQLGetAutoTunningParametersError("Error computing buffer pool parameters")
+
+    def get_max_connections(self) -> int:
+        """Calculate max_connections parameter for the instance."""
+        # Reference: based off xtradb-cluster-operator
+        # https://github.com/percona/percona-xtradb-cluster-operator/blob/main/pkg/pxc/app/config/autotune.go#L61-L70
+
+        bytes_per_connection = 12582912  # 12 Megabytes
```

Comment:
I wonder if this is way too conservative and can still not be enough.

Reply:
Absolutely conservative, it is sort of "safe defaults".
The idea is to provide safe default, and if the user knows the biggest value that does not produce OOM for his particular workload, then he will customize the value via "juju config" (when we have config management).